ui/cli.py

```diff
@@ -0,0 +1,52 @@
+from util.app import App
+import time
+
+
+def console_app(app: App):
+    done = False
+    while not done:
+        try:
+            done = _console_app(app)
+        except EOFError:
+            # Triggered in CLI after starting listen command, then doing Ctrl+C twice to exit program.
+            # This happens without the event queue as well, probably something to do with daemon threads
+            # being blocked/holding onto locks. Is this considered a hack, or is it an acceptable response?
+            done = True
+
+
+def _console_app(app: App):
+    app.print_thread.set_paused()
+
+    print("\nSelect command")
+    print('-' * 30)
+    print("listen\t\t(Use keyboard interrupt to return to menu)")
+    print('quit')
+    print("room-buzz-on")
+    print("room-buzz-off")
+    print("door-light-on")
+    print("door-light-off")
+    print('-' * 30)
+    print('Enter command:', end='')
+    i = input()
+
+    if i == 'room-buzz-on':
+        app.room_buzzer_on()
+    elif i == 'room-buzz-off':
+        app.room_buzzer_off()
+    elif i == 'door-light-on':
+        app.door_light_on()
+    elif i == 'door-light-off':
+        app.door_light_off()
+    elif i == 'listen':
+        app.print_thread.set_unpaused()
+        try:
+            while True:
+                time.sleep(1)
+        except KeyboardInterrupt:
+            print("Stopped listening...")
+    elif i == 'quit':
+        return True
+    else:
+        print("Unknown command")
```

Comment:
Since you've used `match/case` elsewhere in this PR, this would be a good place to utilize it too?

Reply:
It is a little nicer than if statements https://github.com/magley/my-pi-home/pull/4/commits/bf0ddd5d7d8225d6083945886ff7531babfc193a
Be aware though that match in python is a little wonky (for example, we couldn't extract the command strings to constants and match on those without some hackiness). Also, they're >=3.10 only.

Reply:
Hopefully the RPis have >=3.10 installed, but that's something we should've checked last time...

Reply:
Another "cutting-edge" python feature we're using are generic built-in types (for example list[int] instead of typing.List[int]). This is a >=3.9 feature.
A quick google search suggests the default python version on RasPis is 3.9...
Are we going to demonstrate the project on the RasPi for the first control point? That would be problematic also because we haven't actually tested the unsimulated branches of the code work.